include/fcl/narrowphase/narrowphase.h

```diff
@@ -38,32 +38,54 @@
 #ifndef FCL_NARROWPHASE_H
 #define FCL_NARROWPHASE_H
 
+#include <algorithm>
+
+#include "fcl/collision_data.h"
 #include "fcl/narrowphase/gjk.h"
 #include "fcl/narrowphase/gjk_libccd.h"
 
-
-
 namespace fcl
 {
-
-
-
 /// @brief collision and distance solver based on libccd library.
 struct GJKSolver_libccd
 {
   /// @brief intersection checking between two shapes
+  /// @deprecated use shapeIntersect(const S1&, const Transform3f&, const S2&, const Transform3f&, std::vector<ContactPoint>*) const
   template<typename S1, typename S2>
+  FCL_DEPRECATED
   bool shapeIntersect(const S1& s1, const Transform3f& tf1,
                       const S2& s2, const Transform3f& tf2,
-                      Vec3f* contact_points, FCL_REAL* penetration_depth, Vec3f* normal) const
+                      Vec3f* contact_points, FCL_REAL* penetration_depth, Vec3f* normal) const;
+
+  /// @brief intersection checking between two shapes
+  template<typename S1, typename S2>
+  bool shapeIntersect(const S1& s1, const Transform3f& tf1,
+                      const S2& s2, const Transform3f& tf2,
+                      std::vector<ContactPoint>* contacts) const
```

Comment:
It makes sense to me. I can see there are other functions default function parameter such as `shapeTriangleIntersect`, `shapeDistance`, and `shapeTriangleDistance`.

Especially, `shapeDistance(const S1&, const Transform3f&, const S2&, const Transform3f&, FCL_REAL*)` looks redundant once we have `shapeDistance(const S1&, const Transform3f&, const S2&, const Transform3f&, FCL_REAL* = NULL, Vec3f* = NULL, Vec3f* = NULL)`.